### PR TITLE
Removing `Missing CODA_IN_APP_SURVEY_TOKEN` error message for development in Toast

### PR DESCRIPTION
**Pull request by samkevin1**
Related: https://github.com/opencollective/opencollective/issues/7048

### CODE REVIEW

File: server/graphql/v2/mutation/SendSurveyResponseMutation.ts [hunk 1]
```diff
@@ -53,6 +53,10 @@ export const sendSurveyResponseMutation = {
     };
 
     try {
+      if (config.env === 'development') {
+        return true;
+      }
+
```
Comment (Betree) [inline server/graphql/v2/mutation/SendSurveyResponseMutation.ts:59]: This is a blocker to me: we need the ability to work on this feature in local dev environments. Shared my concerns in https://github.com/opencollective/opencollective/issues/7048#issuecomment-1876453672.

File: server/graphql/v2/mutation/SendSurveyResponseMutation.ts [hunk 2]
```diff
@@ -30,8 +30,8 @@ export const sendSurveyResponseMutation = {
     }
     const CODA_TOKEN = process.env.CODA_IN_APP_SURVEY_TOKEN;
 
-    if (!CODA_TOKEN) {
-      throw new UnexpectedError('Missing CODA_IN_APP_SURVEY_TOKEN');
+    if (!CODA_TOKEN && config.env !== 'development') {
+      throw new UnexpectedError('Missing CODA_IN_APP_SURVEY_TOKEN environment variable');
```
Comment (Betree) [inline server/graphql/v2/mutation/SendSurveyResponseMutation.ts:34]: I would argue for the opposite: never show such a technical error message to end-users.

```suggestion
    if (!CODA_TOKEN) {
      throw new UnexpectedError(config.env !== 'development' ? 'Missing CODA_IN_APP_SURVEY_TOKEN environment variable' : 'An unexpected error happened while submitting the survey');
```

**Review by Betree — CHANGES_REQUESTED** [review]
Thanks for the contribution! I have a few concerns with the scope of the root issue that I've shared in https://github.com/opencollective/opencollective/issues/7048#issuecomment-1876453672.

File: server/graphql/v2/mutation/SendSurveyResponseMutation.ts (hunk 2 above)
Comment (samkevin1) [reply server/graphql/v2/mutation/SendSurveyResponseMutation.ts:34]: Shouldn't it be `(config.env === 'development' ? 'Missing CODA_IN_APP_SURVEY_TOKEN environment variable' : 'An unexpected error happened while submitting the survey')` instead?

Comment (Betree) [reply server/graphql/v2/mutation/SendSurveyResponseMutation.ts]: Indeed

**Review by Betree — APPROVED** [review]

**Comment by Betree** [comment]
Good for me, I'll let @gustavlrsn give a 2nd review

**Review by gustavlrsn — APPROVED** [review]
Looks good to me!

**Pull request merged**